upf/plan_validator.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2021 AIPlan4EU project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+from typing import Dict
+import upf.environment
+from upf.simplifier import Simplifier
+from upf.substituter import Substituter
+from upf.problem import Problem
+from upf.plan import Plan, ActionInstance, SequentialPlan
+from upf.fluent import Fluent
+from upf.fnode import FNode
+from upf.expression import Expression
+
+
+class PlanValidator(object):
+    """Performs plan validation."""
+    def __init__(self, env: 'upf.environment.Environment'):
+        self._env = env
+        self.manager = env.expression_manager
+        self.type_checker = env.type_checker
+        self._simplifier = Simplifier(self._env)
+        self._substituter = Substituter(self._env)
+        self._subst: Dict[Expression, Expression] = {}
```

Comment:
```suggestion
        self._assignments: Dict[FNode, FNode] = {}
```

Reply:
Actually it was Dict[FNode, FNode], but mypy doesn't understand it is right.